test/trento/application/process_managers/deregistration_process_manager_test.exs

```diff
@@ -50,6 +65,53 @@ defmodule Trento.DeregistrationProcessManagerTest do
                })
     end
 
+    test "should start the process manager when DatabaseInstanceRegistered event arrives" do
+      host_id = UUID.uuid4()
+
+      assert {:start, ^host_id} =
+               DeregistrationProcessManager.interested?(%DatabaseInstanceRegistered{
+                 host_id: host_id
+               })
+    end
+
+    test "should start the process manager when ApplicationInstanceRegistered event arrives" do
+      host_id = UUID.uuid4()
+
+      assert {:start, ^host_id} =
+               DeregistrationProcessManager.interested?(%ApplicationInstanceRegistered{
+                 host_id: host_id
+               })
+    end
+
+    test "should start process managers when SapSystemRolledUp arrives" do
+      db_host_id_1 = UUID.uuid4()
+      db_host_id_2 = UUID.uuid4()
+      app_host_id_1 = UUID.uuid4()
+      app_host_id_2 = UUID.uuid4()
+
+      database_instances = [
```

Comment:
This looks an antipattern:
Maybe:
```
[%{host_id: db_host_id_1}, %{host_id: db_host_id_2}] = database_instances = build_list(2, :sap_system_instance)
```
And so on. In this case (and almost always when you use factories), giving pre-created unique id-s to the created objects doesn't make sense. Let the factory do that work for you. You can capture later on these values.